mscore/plugin/api/elements.cpp

```diff
@@ -69,6 +69,49 @@ void Note::setTpc(int val)
             set(Pid::TPC2, val);
       }
 
+
+//---------------------------------------------------------
+//   Chord::add
+//---------------------------------------------------------
+
+void Chord::add(Ms::PluginAPI::Element* wrapped)
+      {
+      Ms::Element* s = wrapped->element();
+      if (s)
+            {
+            // Ensure that the object has the expected ownership
+            if (wrapped->ownership() == Ownership::SCORE) {
+                  qWarning("Chord::add: Cannot add this element. The element is already part of the score.");
+                  return;        // Don't allow operation.
+            }
+            // Score now owns the object.
+            wrapped->setOwnership(Ownership::SCORE);
+            // Provide parentage for element.
+            s->setParent(chord());
+            // Create undo op and add the element.
+            chord()->score()->undoAddElement(s);
+            // If a note, ensure the element has proper Tpc values. (Will crash otherwise)
+            if (s->type() == ElementType::NOTE)
+                  toNote(s)->setTpcFromPitch();
```

Comment:
It seems this still doesn't prevent `Chord.add` from crashing MuseScore on trying to add a note. The issue appears when exceprts are present in a score. To reproduce it:
1) Create a score with one note.
2) Create parts (File→Parts→Generate→OK) (test score at this stage: [test.zip](https://github.com/musescore/MuseScore/files/3443641/test.zip), archived as Github doesn't seem to accept .mscx and .qml files).
3) Click on a note and run this plugin (uses the recently merged selection API but it is possible to create a test plugin without it): [testaddnote.zip](https://github.com/musescore/MuseScore/files/3443649/testaddnote.zip).

Result: MuseScore crashes.

It looks like it would be preferable to set the correct TPC values before adding a note to a score. Adding a modified version of this code before the `undoAddElement()` call seems to fix the issue but maybe I am missing something else with this solution:
```
if (s->isNote()) {
      s->setTrack(chord()->track());
      toNote(s)->setTpcFromPitch();
      }
```